Content.Server/Resist/ResistLockerSystem.cs

```diff
@@ -0,0 +1,121 @@
+            TryComp(uid, out EntityStorageComponent? storageComponent);
+            if (!Resolve(uid, ref storageComponent))
+                return;
+
+            if (!component.IsResisting)
+            {
+                if (TryComp<LockComponent>(uid, out var lockComponent) && lockComponent.Locked || storageComponent.IsWeldedShut)
+                {
+                    AttemptResist(args.Entity, uid, storageComponent, component);
+                }
+            }
+
+        }
+        private void AttemptResist(EntityUid user, EntityUid target, EntityStorageComponent? storageComponent, ResistLockerComponent? resistLockerComponent)
+        {
+            if (!Resolve(target, ref storageComponent, ref resistLockerComponent))
+                return;
+
+            resistLockerComponent.CancelToken = new();
+            var doAfterEventArgs = new DoAfterEventArgs(user, resistLockerComponent.ResistTime, resistLockerComponent.CancelToken.Token, target)
+            {
+                BreakOnTargetMove = false,
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                BreakOnStun = true,
+                NeedHand = false, //No hands 'cause we be kickin'
+                TargetFinishedEvent = new ResistDoAfterComplete(user, target),
+                TargetCancelledEvent = new ResistDoAfterCancelled(user)
+            };
+
+            resistLockerComponent.IsResisting = true;
+            _popupSystem.PopupEntity(Loc.GetString("resist-locker-component-start-resisting"), user, Filter.Entities(user));
+            _doAfterSystem.DoAfter(doAfterEventArgs);
+        }
+
+        private void OnDoAfterComplete(EntityUid uid, ResistLockerComponent component, ResistDoAfterComplete ev)
+        {
+            component.IsResisting = false;
+
+            if (TryComp<EntityStorageComponent>(uid, out var storageComponent))
+            {
+                if (storageComponent.IsWeldedShut)
+                    storageComponent.IsWeldedShut = false;
+
+                if (TryComp<LockComponent>(ev.Target, out var lockComponent))
+                {
+                    lockComponent.Locked = false;
+                }
+
+                component.CancelToken = null;
+                storageComponent.TryOpenStorage(ev.User);
+            }
+        }
+
+        private void OnDoAfterCancelled(EntityUid uid, ResistLockerComponent component, ResistDoAfterCancelled ev)
+        {
+            component.IsResisting = false;
+            component.CancelToken = null;
+            _popupSystem.PopupEntity(Loc.GetString("resist-locker-component-resist-interrupted"), ev.User, Filter.Entities(ev.User));
+        }
+
+        private void OnRemovedFromContainer(EntityUid uid, ResistLockerComponent component, EntRemovedFromContainerMessage message)
+        {
+            if (component.CancelToken != null)
+            {
+                component.IsResisting = false;
+                component.CancelToken.Cancel();
+            }
```

Comment:
```suggestion
            component.IsResisting = false;
            component.CancelToken?.Cancel();
            component.CancelToken = null;
```

Reply:
Although technically IsResisting should get handelled by the cancelled method above and you can probably remove that and just have CancelToken?.Cancel() maybe, would need to check in-game.

Reply:
Checked in-game and IsResisting flips false through the cancelled method and so i've removed it from OnRemovedFromContainer